4.5. Set-Up GitHub repository

JaleneA opened the pull request:
**Basic Set-Up For FFM Application** 

**Development Stack**
- React Native (Frontend)
- Flask (Backend)

The readme outlines the steps in running the project.

JaleneA committed:
Minor Cleanup To Start

- Including two new requirements to requirements.txt

- Removed unneeded text/code within wsgi.py and readme.md etc

JaleneA committed:
File Rearrangement

Moved, removed and configured files adjusting for a different file setup. Code initializes and runs as expected

JaleneA committed:
React Native Frontend + Flask Backend

Working on implementing a React Native Frontend then connecting it with the Flask Backend, currently they are linked as the data stored within the flask temp db can be fetched via API on the frontend.

Need to look over the ports and host IPs used, as I'm currently not sure if it'll be the same for each person.

Current state of the application was tested using a physical device (android phone) using Expo Go and an Android Emulator via Android Studio

JaleneA committed:
Updated readme.md

Modified the readme removing irrelevant text and including the (current) process of running the project.

JaleneA commented:
![](https://github.trello.services/images/mini-trello-icon.png) [4.4. SetUp Github Repository](https://trello.com/c/3S63yaqm/19-44-setup-github-repository)

JaleneA committed:
Updated dev.yml

Commented out command that runs tests
Updated path to suit in other areas for futures

JaleneA commented:
![](https://github.trello.services/images/mini-trello-icon.png) [4.4. SetUp Github Repository](https://trello.com/c/3S63yaqm/19-44-setup-github-repository)

RynniaRyan commented:
Very cool, I got it to run but I had to change some of the commands and do additional commands to get it working for some reason, may have to change the read.me file because of this?

## 1. For Running Flask (Backend):
Below I could not get `flask run` to work and got this error message:
![image](https://github.com/user-attachments/assets/c502847d-09f8-4343-b973-213aac6b366f)

So I used `flask run --host=0.0.0.0` instead 


## 2. For Starting React:
I couldn't get to start up the web version because the option was greyed out

So I had to do `npx expo install react-dom react-native-web @expo/metro-runtime` and it works! 
![image](https://github.com/user-attachments/assets/b624ee02-5e0e-4768-9559-11740937021a)


I thought that would have been included within the `npm install -g expo-cli` command but ig not

JaleneA commented:
> Very cool, I got it to run but I had to change some of the commands and do additional commands to get it working for some reason, may have to change the read.me file because of this?
> 
> ## 1. For Running Flask (Backend):
> Below I could not get `flask run` to work and got this error message: ![image](https://private-user-images.githubusercontent.com/122883419/408862369-c502847d-09f8-4343-b973-213aac6b366f.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mzg0OTY2MzEsIm5iZiI6MTczODQ5NjMzMSwicGF0aCI6Ii8xMjI4ODM0MTkvNDA4ODYyMzY5LWM1MDI4NDdkLTA5ZjgtNDM0My1iOTczLTIxM2FhYzZiMzY2Zi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwMjAyJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDIwMlQxMTM4NTFaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT02NTFlNmUxYmE4ZmFmYzQwMjZhZGRlZTUxOTU0ZGI3MTg5M2MyN2Y3ZWQ1ZmQxNmFiNzViYzUxZTU1NDBhOWVkJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.ioOHXIibg_Rxolr-xAMGn_NKHKGyKHO6iJmNhuICX-Y)
> 
> So I used `flask run --host=0.0.0.0` instead ![image](https://private-user-images.githubusercontent.com/122883419/408862301-401553e9-cb07-46c7-bf36-807f7cf416a4.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mzg0OTY2MzEsIm5iZiI6MTczODQ5NjMzMSwicGF0aCI6Ii8xMjI4ODM0MTkvNDA4ODYyMzAxLTQwMTU1M2U5LWNiMDctNDZjNy1iZjM2LTgwN2Y3Y2Y0MTZhNC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwMjAyJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDIwMlQxMTM4NTFaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT00MzdhZWUzZGFhYzU0Y2ZkZDQyMWEyYWVkNzNkZDg0Mjg4MzBlMWE5NmY0NWQ4MDRjNDk2MGVhYWExNDVjY2RiJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.TYDnGPKVAANKoIGsrUerfc1Sa97rZqfdpbqC8RkVEPU)
> 
> ## 2. For Starting React:
> I could'nt get to start up the web version because the option was greyed out: ![image](https://private-user-images.githubusercontent.com/122883419/408862583-71025e80-ccb4-4e8b-a4c5-8440f0ec9449.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mzg0OTY2MzEsIm5iZiI6MTczODQ5NjMzMSwicGF0aCI6Ii8xMjI4ODM0MTkvNDA4ODYyNTgzLTcxMDI1ZTgwLWNjYjQtNGU4Yi1hNGM1LTg0NDBmMGVjOTQ0OS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwMjAyJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDIwMlQxMTM4NTFaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0wOTEyYjFmZDhkN2MzNjA2ZjM3ODkzMmJhNGIyZDNhODMyMjBmMTgwMzRiNzQ5YTM0ZDJlMmI4M2VlNjNkY2VhJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.Ebn9gaMZNOdZnQXn2XdYUWxSGk0jitiCFKD_9OqQPwk)
> 
> So I had to do `npx expo install react-dom react-native-web @expo/metro-runtime` and it works! ![image](https://private-user-images.githubusercontent.com/122883419/408863011-b624ee02-5e0e-4768-9559-11740937021a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mzg0OTY2MzEsIm5iZiI6MTczODQ5NjMzMSwicGF0aCI6Ii8xMjI4ODM0MTkvNDA4ODYzMDExLWI2MjRlZTAyLTVlMGUtNDc2OC05NTU5LTExNzQwOTM3MDIxYS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwMjAyJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDIwMlQxMTM4NTFaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT05ZjgwOTE0OTk1NTMxYWZjYTMxN2Y1MWY4YjUxZjc0ZDhlYTNlMjZjZjlhOWQxNDY1YWY5NTBhMGQ1ODA2NzA5JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.T7yN8oXTffaWu--6Z-e2eIh_0Oc44BV5XbwMzp-L8Ys)
> 
> I thought that would have been included within the `npm install -g expo-cli` command but ig not

RynniaRyan reviewed (approved):
**Approved 👍**

JaleneA commented:
@RynniaRyan 
1. Yeah you would have needed to change the address to suit your device, because I had it set to mines, having it open on all addresses was unnerving to say the least.
2. Yeah, that's why in the readme.md I specifically said that you can run it on an emulator or a physical device, we don't need the web version because we're developing a mobile application.

RynniaRyan commented:
> 1. Yeah you would have needed to change the address to suit your device, because I had it set to mines, having it open on all addresses was unnerving to say the least.

@JaleneA 😭 hmm yh you're right, i'll try that instead then, also I don't have merge privileges (write access)

JaleneA committed:
Template Integration

Integrated a free login/signup template from React Native Market, involved careful moifications as the template uses old versions or version not compatible with current project.

Attempted to make host addresses more flexible for other members.

Through integrating the template, learned how to make a web version available via localhost, so now both an android version and web version is valid

Flask API still works for both web and android

readme.md updated to suit now with visuals :)

RynniaRyan reviewed (approved):
The new template integration seems to be working pretty smooth also I "started over" my set-up in testing this and it was better this time. Expo App method works fine for me. This is very cool

**Approved 👍**

JaleneA commented:
![](https://github.trello.services/images/mini-trello-icon.png) [4.5. SetUp Github Repository](https://trello.com/c/3S63yaqm/19-45-setup-github-repository)
